### PR TITLE
NICPS-525: Fixed multi sms issue

### PR DESCRIPTION
**Pull request by deepakzimb**
Bugfix for repeating message on same mobile number
Bugfix for repeating message on same mobile number when email id added with mobile number

### CODE REVIEW

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 1]
```diff
@@ -71,13 +74,16 @@ private void sendEmailSMS(Message zMsg) throws ServiceException,MessagingExcepti
 			fullMessage = fullMessage + convertStringToUnicode("\n");
 			fullMessage = fullMessage + convertStringToUnicode(bodyMsg);
 		}
+		Set<String> uniqueMobileSet = new HashSet<String>();
 		for (int i = 0; i < recipients.length; i++) {
 			String rcptAddresses = recipients[i].toString();
 			String[] splitMobNumber = rcptAddresses.indexOf("<") < 0 ? rcptAddresses.split("@")
 					: rcptAddresses.split("<")[1].replaceAll(">", "").split("@");
 			String mobNumber = splitMobNumber[0];
+			String mobNumberWithoutCountryCode = splitMobNumber[0].replace("+", "");
```
[the remark this answers is not in the document]
Comment (deepakzimb) [reply store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: John, we had discussion that SMS API accepting any number so we are not doing validation on minimum and maximum number.

[the remark this answers is not in the document]
Comment (deepakzimb) [reply store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: We are allowing user to add + and country code and user can add mobile number without country code also.

[the remark this answers is not in the document]
Comment (deepakzimb) [reply store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: We are sending mobile number with country code to SMS API

[the remark this answers is not in the document]
Comment (jeastman) [reply store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]: Understand that we are not validating the number and we are letting the SMS Gateway do the validation.

File: store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java [hunk 2]
```diff
@@ -71,13 +73,16 @@ private void sendEmailSMS(Message zMsg) throws ServiceException,MessagingExcepti
 			fullMessage = fullMessage + convertStringToUnicode("\n");
 			fullMessage = fullMessage + convertStringToUnicode(bodyMsg);
 		}
+		Set<String> uniqueMobileSet = new HashSet<String>();
 		for (int i = 0; i < recipients.length; i++) {
 			String rcptAddresses = recipients[i].toString();
 			String[] splitMobNumber = rcptAddresses.indexOf("<") < 0 ? rcptAddresses.split("@")
 					: rcptAddresses.split("<")[1].replaceAll(">", "").split("@");
 			String mobNumber = splitMobNumber[0];
+			String mobNumberWithoutSymbol = splitMobNumber[0].replace("+", "");
```
Comment (jkester1986) [inline store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java]:
```suggestion
			String mobNumberWithoutSymbol = splitMobNumber[0].replace(/[()\-+. ]/g, '');
```

**Pull request merged**